pythx/cli/utils.py

```diff
@@ -0,0 +1,247 @@
+    :param exit_on_missing: Return if the file is missing
+    :param no_cache: Disable the API result cache
+    :return:
+    """
+    if not path.isfile(config_path):
+        if exit_on_missing:
+            return None
+        # config doesn't exist - assume first use
+        eth_address = environ.get("PYTHX_USERNAME") or click.prompt(
+            "Please enter your Ethereum address",
+            type=click.STRING,
+            default="0x0000000000000000000000000000000000000000",
+        )
+        password = environ.get("PYTHX_PASSWORD") or click.prompt(
+            "Please enter your MythX password",
+            type=click.STRING,
+            hide_input=True,
+            default="trial",
+        )
+        c = Client(
+            eth_address=eth_address,
+            password=password,
+            staging=staging,
+            no_cache=no_cache,
+        )
+        c.login()
+        update_config(config_path=config_path, client=c)
+    else:
+        config = parse_config(config_path, tokens_required=True)
+        c = Client(
+            eth_address=config["username"],
+            password=config["password"],
+            access_token=config["access"],
+            refresh_token=config["refresh"],
+            staging=staging,
+            no_cache=no_cache,
+        )
+    return c
+
+
+def ps_core(config, staging, number):
+    """A helper method to retrieve data from the analysis list endpoint.
+
+    This functionality is used in the :code:`pythx ps`, as well as the :code:`pythx top`
+    subcommands.
+
+    :param config: The configuration file's path
+    :param staging: Boolean to denote whether to use the MythX staging deployment
+    :param number: The number of analyses to retrieve
+    :return: The API response as AnalysisList domain model
+    """
+    c = recover_client(config_path=config, staging=staging)
+    if c.eth_address == "0x0000000000000000000000000000000000000000":
+        click.echo(
+            (
+                "This functionality is only available to registered users. "
+                "Head over to https://mythx.io/ and register a free account to "
+                "list your past analyses. Alternatively, you can look up the "
+                "status of a specific job by calling 'pythx status <uuid>'."
+            )
+        )
+        sys.exit(0)
+    resp = c.analysis_list()
+    # TODO: paginate if too few analyses
+    resp.analyses = resp.analyses[: number + 1]
+    update_config(config_path=config, client=c)
+    return resp
+
+
+def get_source_location_by_offset(filename, offset):
+    """Retrieve the Solidity source file's location based on the source map offset.
+    :param filename: The Solidity file to analyze
+    :param offset: The source map's offset
+    :return: The line number
+    """
+    overall = 0
+    line_ctr = 0
+    if not path.exists(filename):
+        LOGGER.warning("Could not find file %s", filename)
```

Comment:
Add `raise FileNotFoundError("Could not find file %s".format(filename))`

Reply:
When `open()` fails to find a file, it automatically raises a `FileNotFoundError`. So if the path doesn't exist, we log the warning and catch it in the functions using `get_source_location_by_offset`

Reply:
Oh ok!